### DISCUSSION TITLE
Set `readPreference` to `primary` by default for MongoDB EntityManager transactionals

### DISCUSSION THREAD

**Discussion by snigdha920** [mikro-orm/mikro-orm#4491] [Ideas]
For a MongoDB transaction, the read preference can only be set to `primary`. Here's [why](https://www.mongodb.com/community/forums/t/why-can-t-read-preference-be-secondary-in-a-transaction/204432) 

This means that we need to remember to set `readPreference` to `primary` in every transactional we write, otherwise it will error out with: 

```
DriverException: Read preference in a transaction must be primary, not: secondaryPreferred
```

I was thinking we can set this option on the transactional of the mongodb EntityManager itself so we don't need to manually specify it everywhere. 

A downside could be that the developers can be unaware that this read preference is being set for mongodb, which can catch them by surprise while monitoring primary/secondary nodes. We can update the documentation to reflect this change if we make it, perhaps a small note here: https://mikro-orm.io/docs/transactions#approach-2-explicitly

Let me know what you think, I can open a PR for this with some guidance.

**Comment by B4nan** [comment]
We don't set this value anyhow, and I'd guess the defaults just work (as all the tests just work). Sounds like you are overriding that value somewhere yourself globally?

Btw what's the point of using transactions for read operations? I mean why would you even end up in this situation? It's weird that they even have this option there, if the only allowed value is `primary`...
